src/main/java/com/exadel/discount/service/impl/CityServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+package com.exadel.discount.service.impl;
+
+import com.exadel.discount.dto.CityDTO;
+import com.exadel.discount.entity.City;
+import com.exadel.discount.exception.NotFoundException;
+import com.exadel.discount.mapper.CityMapper;
+import com.exadel.discount.repository.CityRepository;
+import com.exadel.discount.service.CityService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CityServiceImpl implements CityService {
+
+    private final CityRepository cityRepository;
+    private final CityMapper cityMapper;
+
+    @Override
+    public List<CityDTO> findAllCities() {
+        log.debug("Getting list of all Cities");
+
+        List<CityDTO> cityDTOList = cityMapper.getListDTO(cityRepository.findAll());
+
+        log.debug("Successfully got list of all Cities");
+        return cityDTOList;
+    }
+
+    @Override
+    public List<CityDTO> findAllCitiesByCountryId(UUID id) {
+        log.debug("Getting list of all Cities by CountryId");
+
+        List<CityDTO> cityDTOList = cityMapper.getListDTO(cityRepository.findAll()
+                .stream()
+                .filter(s -> s.getCountry().getId().equals(id))
+                .collect(Collectors.toList()));
```

Comment:
It should be in repository. Add method findByCountry()

Reply:
> 
> 
> It should be in repository. Add method findByCountry()

Done